oeps/best-practices/oep-0011/decisions/0007-typescript.rst

```diff
@@ -0,0 +1,43 @@
+Use TypeScript for static type checking
+#######################################
+
+Status
+******
+
+Accepted
+
+Context
+*******
+
+As documented in :doc:`OEP-0011-0004 <0004-js-ecma>`, the Open edX community has previously 
+:ref:`decided against adopting TypeScript <Reasons for rejecting TypeScript>`. In the time 
+since that decision was made, TypeScript has grown to be quite popular and commonplace for JavaScript developers.
+This has led to more discussion within the community about the `pros and cons of adopting TypeScript`_.
+
+Decision
+********
+
+Frontend repositories within the Open edX project will be updated to support the use of TypeScript, without
+requiring existing frontend JavaScript code be rewritten in TypeScript.
```

Comment:
Reading this (the Decision & Consequence sections) I lack context as to _why_ this decision has been made and what consequence it has (like, what's the benefit of using TypeScript). I know you link out to external resources but I think it could be beneficial to add a sentence or two to provide context without requiring reading through external resources as well.

Reply:
I added a quick summary of a couple benefits of typescript here https://github.com/openedx/open-edx-proposals/compare/3af815b84c789af807d4e42927938024557c669f..f71d62cd46797a78be384e73f6577913ea599014

Let me know if there's something you feel is still missing!

Reply:
Love it, thank you!